backend/plonk/bn254/solidity.go

```diff
@@ -55,59 +56,49 @@ library Utils {
         res[i] = uint8(b1[i]);
         ++i;
       }
-    }
 
-    bytes32 xorTmp = b0 ^ b1;
-    tmp = abi.encodePacked(xorTmp, uint8(2), dst, sizeDomain);
+      tmp = abi.encodePacked(b0 ^ b1, uint8(2), dst, sizeDomain);
 
-    b1 = sha256(tmp);
+      b1 = sha256(tmp);
 
-    unchecked {
       // TODO handle the size of the dst (check gnark-crypto)
       for (uint i; i < 16; ) {
         res[i + 32] = uint8(b1[i]);
         ++i;
       }
     }
-
-    return res;
   }
 
-
   /**
   * @dev cf https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-06#section-5.2
   * corresponds to https://github.com/ConsenSys/gnark-crypto/blob/develop/ecc/bn254/fr/element.go
   */
   function hash_fr(uint256 x, uint256 y) internal pure returns (uint256 res) {
     // interpret a as a bigEndian integer and reduce it mod r
-    uint8[48] memory xmsg = expand_msg(x, y);
-    // uint8[48] memory xmsg = [0x44, 0x74, 0xb5, 0x29, 0xd7, 0xfb, 0x29, 0x88, 0x3a, 0x7a, 0xc1, 0x65, 0xfd, 0x72, 0xce, 0xd0, 0xd4, 0xd1, 0x3f, 0x9e, 0x85, 0x8a, 0x3, 0x86, 0x1c, 0x90, 0x83, 0x1e, 0x94, 0xdc, 0xfc, 0x1d, 0x70, 0x82, 0xf5, 0xbf, 0x30, 0x3, 0x39, 0x87, 0x21, 0x38, 0x15, 0xed, 0x12, 0x75, 0x44, 0x6a];
+    unchecked {
+      uint8[48] memory xmsg = expand_msg(x, y);
+      // uint8[48] memory xmsg = [0x44, 0x74, 0xb5, 0x29, 0xd7, 0xfb, 0x29, 0x88, 0x3a, 0x7a, 0xc1, 0x65, 0xfd, 0x72, 0xce, 0xd0, 0xd4, 0xd1, 0x3f, 0x9e, 0x85, 0x8a, 0x3, 0x86, 0x1c, 0x90, 0x83, 0x1e, 0x94, 0xdc, 0xfc, 0x1d, 0x70, 0x82, 0xf5, 0xbf, 0x30, 0x3, 0x39, 0x87, 0x21, 0x38, 0x15, 0xed, 0x12, 0x75, 0x44, 0x6a];
 
-    // reduce xmsg mod r, where xmsg is intrepreted in big endian
-    // (as SetBytes does for golang's Big.Int library).
-    for (uint i; i < 32; ) {
-      res += uint256(xmsg[47 - i]) << (8 * i);
-      unchecked {
-        i++;
+      // reduce xmsg mod r, where xmsg is intrepreted in big endian
+      // (as SetBytes does for golang's Big.Int library).
+      for (uint i; i < 32; ) {
+        res += uint256(xmsg[47 - i]) << (8 * i);
```

Comment:
@ThomasPiellard I have included this in the unchecked behaviour - we will validate on UAT